src/server/game/Maps/Map.cpp

```diff
@@ -753,8 +753,11 @@ void Map::Update(uint32 t_diff)
         {
             //player->Update(t_diff);
             WorldSession* session = player->GetSession();
+
             MapSessionFilter updater(session);
             session->Update(t_diff, updater);
+
+            session->UpdateForPlayersOnMap(t_diff);
```

Comment:
Isn't the point of updater/filter argument on WorldSession::Update to do exactly this? Execute code based on update context

Reply:
i could move the code inside WorldSession::UpdateForPlayersOnMap to WorldSession::Update but I would have to use a dynamic cast on the `updater` reference:
````C++
try
{
    dynamic_cast<MapSessionFilter&>(updater);
    // Send time sync packet every 10s.
    if (_timeSyncTimer > 0)
    {
        if (diff >= _timeSyncTimer)
            SendTimeSync();
        else
            _timeSyncTimer -= diff;
    }
}
catch (std::bad_cast exp)
{
}
````

Wouldn't that have a noticable performance impact?

Reply:
You don't need to rely on dynamic_cast, you can place the code under `updater.ProcessLogout()` - thats the other function of the filter that determines context

Reply:
Done. In that case, the virtual function `ProcessLogout` should probably be renamed into something more meaningful since i'm using it out of the log-out context.

Reply:
Yes, I already renamed it long time ago in master branch https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Server/WorldSession.h#L848

Reply:
Done